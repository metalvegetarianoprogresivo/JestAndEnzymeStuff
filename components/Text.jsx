import React from 'react';

class Text extends React.Component {

    render() {
        const { text } = (this.props) ? this.props : null;
        const { textId } = (this.props) ? this.props : 'h1_text_id';

        return (
            <div>
                <h1 id={ textId }>{text}</h1>
            </div>
        );
    }
}

export default Text;
