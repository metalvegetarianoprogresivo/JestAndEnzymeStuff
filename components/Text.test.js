import 'jsdom-global/register'
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Text from '../components/Text';

Enzyme.configure({ adapter: new Adapter() });

describe('Timeline', () => {
    let wrapper;

    it('renders correctly', () => {
        const tree = renderer
            .create(<Text text="Hola Mundo" />)
            .toJSON();
    
        expect(tree).toMatchSnapshot();
    });

    it('has a title of Text', () => {
        wrapper = mount(<Text text="hola" textId="holamorro" />)
        expect(wrapper.find('#holamorro').text()).toBe("hola")
    });

    it('wraps content in a div with #holamorro id', () => {
        wrapper = shallow(<Text text="hola" textId="holamorro" />);
        expect(wrapper.find('#holamorro').length).toEqual(1);
    });
});
