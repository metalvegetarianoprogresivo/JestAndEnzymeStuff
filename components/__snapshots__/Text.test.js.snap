// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
<div>
  <h1>
    Hola Mundo
  </h1>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <h1>
    Hola Mundo
  </h1>
</div>
`;
